### ISSUE TITLE
Import statement deleted when reformatting multiline boolean condition

### ISSUE THREAD

**Issue by mhernand40** [pinterest/ktlint#1215]
Take the following two classes:
```Kotlin
package com.repro.internal

class Foo

```
```Kotlin
package com.repro

import com.repro.internal.Foo

class Repro {
    fun repro(condition: Boolean, obj: Any): Boolean {
        return obj is Foo
            && condition
    }
}
```

## Expected Behavior
When running the formatter, the `Repro` class should be formatted as such:
```Kotlin
package com.repro

import com.repro.internal.Foo

class Repro {
    fun repro(condition: Boolean, obj: Any): Boolean {
        return obj is Foo && condition
    }
}
```

## Observed Behavior
When running the formatter, the `Repro` class gets formatted as such, with the `import` statement removed:
```Kotlin
package com.repro

class Repro {
    fun repro(condition: Boolean, obj: Any): Boolean {
        return obj is Foo &&
            condition
    }
}
```

Although the formatter "succeeds", it ends up introducing a compilation error since `Foo` can no longer be resolved. This does not occur in `0.41.0`.

## Your Environment

* Version of ktlint used: 0.42.1
* Version of Gradle used (if applicable):
```
------------------------------------------------------------
Gradle 7.1.1
------------------------------------------------------------

Build time:   2021-07-02 12:16:43 UTC
Revision:     774525a055494e0ece39f522ac7ad17498ce032c

Kotlin:       1.4.31
Groovy:       3.0.7
Ant:          Apache Ant(TM) version 1.10.9 compiled on September 27 2020
JVM:          11.0.9 (Eclipse OpenJ9 openj9-0.23.0)
OS:           Mac OS X 10.16 x86_64
```
* Operating System and version: MacOS Big Sur 11.5

**Comment by paul-dingemans** [comment]
Most likely this issue is solved by #1201 which is already fixed. Can you please retry with latest snapshot version?

**Comment by romtsn** [comment]
Nope, this is still an issue, present on `master`
